chore: refactor plugin-help implementation

shrutiburman opened the pull request:
<!--
We appreciate the effort for this pull request but before that please make sure you read the contribution guidelines, then fill out the blanks below.

Please format the PR title appropriately based on the type of change:
  <type>[!]: <description>
Where <type> is one of: docs, chore, feat, fix, test.
Add a '!' after the type for breaking changes (e.g. feat!: new breaking feature).

**All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.**

Please enter each Issue number you are resolving in your PR after one of the following words [Fixes, Closes, Resolves]. This will auto-link these issues and close them when this PR is merged!
e.g.
Fixes #1
Closes #2
-->

# Fixes [DII-549](https://issues.corp.twilio.com/browse/DII-549)
Also, verifies [DII-496](https://issues.corp.twilio.com/browse/DII-496)

This PR:
Refactors the plugin-help original implementation to 1. separate out optional and required flags 2. hyperlinks to the documentation for each cmd. 

Example of a cmd which only has optional flags: 
![image](https://user-images.githubusercontent.com/87537688/187089148-9a082652-f6fc-4a44-830a-0f1ff19c419a.png)
Example of a cmd which has required & optional flags: 
![Screenshot 2022-08-28 at 11 59 33 PM](https://user-images.githubusercontent.com/87537688/187089267-e36b9901-ffbe-4c20-ae6b-60df2082e5df.png)




### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

shrutiburman commented on src/services/twilio-help/twilio-command-help.js:
```diff
@@ -47,77 +125,10 @@ class TwilioCommandHelp extends CommandHelp.default {
-          left = label.join(', ');
-        }
-        if (flag.type === 'option') {
-          let value = flag.helpValue || flag.name;
```
[self] 
The earlier output was: **`-p, --profile=profile`**
Current output is: **`-p, --profile=<value>`** 
This line is [modified](https://github.com/oclif/core/blob/29f76feb04e067b084009946e54cd840da683932/src/help/command.ts#L269) in oclif/core to show the flag name only when the optn `showFlagNameInTitle` is set.

shrutiburman commented on test/services/twilio-help/twilio-help.test.js:
```diff
@@ -50,15 +50,13 @@ const testHelp = test
 
   testHelp.commandHelp([]).it('Should not display required flags if not configured', (ctx) => {
     expect(ctx.commandHelp).to.contain('api:testHelp');
-    expect(ctx.commandHelp).to.contain('OPTIONS');
```
[self] There's change in the new format. It no longer says "OPTIONS" before listing out flags. 
It can be added based on feedback. 

<img width="834" alt="image" src="https://user-images.githubusercontent.com/87537688/187089932-5e12e604-66d6-4c28-b14b-835873fa2101.png">

kridai replied:
I agree its not needed since we are already dividing options based on required/ optional flags.

kridai reviewed (approved):
LGTM!

charan678 commented:
You have to remove plugin-help from package.json

shrutiburman commented:
> You have to remove plugin-help from package.json

Addressed, also removed oclif/command changes as they would be addressed as in Charan's ticket later.
